core/src/main/java/com/netflix/conductor/core/execution/tasks/Terminate.java

```diff
@@ -81,6 +81,9 @@ public static Boolean validateInputStatus(String status) {
     private void setWorkflowOutput(Map<String, Object> taskOutput, Workflow workflow) {
         if(!taskOutput.isEmpty()) {
             workflow.setOutput(taskOutput);
+			if (workflow.getWorkflowDefinition() != null && (workflow.getWorkflowDefinition().getOutputParameters() != null && !workflow.getWorkflowDefinition().getOutputParameters().isEmpty())) {
```

Comment:
Changing the workflow definition within the workflow does not seem to be the best way to achieve this. This makes the behavior seem like a side-effect rather than intentional.

Reply:
Agreed, but when called updateWorkflowOutput() in deciderService, Checking the workflowDefintion's outputParameters rather than workflow output which has changed in Terminate.java.

Reply:
Removed the code change which modifies workflow definition and added check in DeciderService which seems the better way to fix the bug.